src/displayapp/screens/Timer.cpp

```diff
@@ -1,173 +1,123 @@
+  lv_obj_set_style_local_value_str(buttons[secDown], LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "-");
+
   if (timerController.IsRunning()) {
-    lv_label_set_text(txtPlayPause, Symbols::pause);
+    SetButtonsHidden(true);
+    lv_obj_set_style_local_value_str(btnPlayPause, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, Symbols::pause);
   } else {
-    lv_label_set_text(txtPlayPause, Symbols::play);
-    createButtons();
+    lv_obj_set_style_local_value_str(btnPlayPause, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, Symbols::play);
   }
-  
+  Refresh();
 }
 
 Timer::~Timer() {
   lv_obj_clean(lv_scr_act());
-  
+}
+
+void Timer::UpdateTime() {
+  lv_label_set_text_fmt(time, "%02d:%02d", minutes, seconds);
 }
 
 bool Timer::Refresh() {
   if (timerController.IsRunning()) {
-    uint32_t seconds = timerController.GetTimeRemaining() / 1000;
-    lv_label_set_text_fmt(time, "%02lu:%02lu", seconds / 60, seconds % 60);
+    remainingTime = timerController.GetTimeRemaining() / 1000 + 1;
+    if (remainingTime.IsUpdated()) {
+      minutes = remainingTime.Get() / 60;
+      seconds = remainingTime.Get() % 60;
+      UpdateTime();
+    }
   }
   return running;
 }
 
+void Timer::SetButtonsHidden(bool hidden) {
+  for (uint8_t i = 0; i < 4; i++) {
+    lv_obj_set_hidden(buttons[i], hidden);
+  }
+}
+
 void Timer::OnButtonEvent(lv_obj_t* obj, lv_event_t event) {
-  if (event == LV_EVENT_CLICKED) {
-    if (obj == btnPlayPause) {
-      if (timerController.IsRunning()) {
-        lv_label_set_text(txtPlayPause, Symbols::play);
-        uint32_t seconds = timerController.GetTimeRemaining() / 1000;
-        minutesToSet = seconds / 60;
-        secondsToSet = seconds % 60;
-        timerController.StopTimer();
-        createButtons();
-        
-      } else if (secondsToSet + minutesToSet > 0) {
-        lv_label_set_text(txtPlayPause, Symbols::pause);
-        timerController.StartTimer((secondsToSet + minutesToSet * 60) * 1000);
-        
-        lv_obj_del(btnSecondsDown);
-        btnSecondsDown = nullptr;
-        lv_obj_del(btnSecondsUp);
-        btnSecondsUp = nullptr;
-        lv_obj_del(btnMinutesDown);
-        btnMinutesDown = nullptr;
-        lv_obj_del(btnMinutesUp);
-        btnMinutesUp = nullptr;
-        
-      }
+  if (event != LV_EVENT_CLICKED) {
+    return;
+  }
+
+  if (obj == btnPlayPause) {
+    if (timerController.IsRunning()) {
+      lv_obj_set_style_local_value_str(btnPlayPause, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, Symbols::play);
+      timerController.StopTimer();
+      SetButtonsHidden(false);
+    } else if (seconds + minutes > 0) {
+      lv_obj_set_style_local_value_str(btnPlayPause, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, Symbols::pause);
+      timerController.StartTimer((seconds + minutes * 60) * 1000);
+      SetButtonsHidden(true);
+    }
+  } else if (obj == buttons[minUp]) {
+    minutes = (minutes + 1) % 61;
+  } else if (obj == buttons[minDown]) {
+    if (minutes == 0) {
```

Comment:
I think the same formula as for minUp could be used here: `minutes = (minutes - 1) % 61;`
Also, I think timer can go without any memory/display/count issue up to 99:59. Though maybe its better this way, because its easy to set one hour timer which is probably used quite often.

Reply:
(- 1 % 61) on an unsigned 8 bit int is 11 and not 60. I don't want to change the behaviour much in this PR, but I remember trying to make it count higher some time ago, and having issues with it going above ~70 minutes for some reason.